admin/tests/integration/components/confirm-popup-test.js

```diff
@@ -1,22 +1,80 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { click, render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
 
 module('Integration | Component | confirm-popup', function(hooks) {
   setupRenderingTest(hooks);
 
-  test('it renders', async function(assert) {
-    // Set any properties with this.set('myProperty', 'value');
-    // Handle any actions with this.set('myAction', function(val) { ... });
-    this.set('display', true);
-    this.set('actionConfirm', () => {
-    });
-    this.set('actionCancel', () => {
-    });
+  hooks.beforeEach(function() {
+    this.display = true;
+  });
 
-    await render(hbs`{{confirm-popup show=display confirm=actionConfirm cancel=actionCancel }}`);
+  test('should open confirm', async function(assert) {
+    await render(hbs`<ConfirmPopup @show={{this.display}} />`);
 
     assert.dom('.modal-dialog').exists();
   });
+
+  test('should call cancel action on click on cancel button', async function(assert) {
+    this.cancel = sinon.stub();
+
+    await render(hbs`<ConfirmPopup @show={{this.display}} @cancel={{this.cancel}} />`);
+    await click('button.btn-secondary');
+
+    assert.ok(this.cancel.called);
+    assert.dom('.modal-dialog').doesNotExist();
+  });
+
+  test('should call confirm action on click on confirm button', async function(assert) {
+    this.confirm = sinon.stub();
+
+    await render(hbs`<ConfirmPopup @show={{this.display}} @confirm={{this.confirm}} />`);
+    await click('button.btn-primary');
+
+    assert.ok(this.confirm.called);
+  });
+
+  test('should display default title if it is not defined', async function(assert) {
+    await render(hbs`<ConfirmPopup @show={{this.display}} />`);
+
+    assert.contains('Merci de confirmer');
+  });
+
+  test('should display title in parameter if it is defined', async function(assert) {
+    this.title = 'Titre de test';
+
+    await render(hbs`<ConfirmPopup @show={{this.display}} @title={{this.title}} />`);
+
+    assert.contains(this.title);
+  });
+
+  test('should display default closeTitle if it is not defined', async function(assert) {
+    await render(hbs`<ConfirmPopup @show={{this.display}} />`);
+
+    assert.contains('Annuler');
+  });
+
+  test('should display closeTitle in parameter if it is defined', async function(assert) {
+    this.closeTitle = 'Titre du bouton d\'annulation';
+
+    await render(hbs`<ConfirmPopup @show={{this.display}} @title={{this.closeTitle}} />`);
+
+    assert.contains(this.closeTitle);
+  });
+
+  test('should display default submitTitle if it is not defined', async function(assert) {
+    await render(hbs`<ConfirmPopup @show={{this.display}} />`);
+
+    assert.contains('Confirmer');
+  });
+
+  test('should display submitTitle  in parameter if it is defined', async function(assert) {
```

Comment:
Quel est la différence avec le test de la ligne 59 ?

Reply:
Le test de la ligne 59 concerne le texte du bouton d'annulation, celui-ci concerne le texte du bouton de confirmation.

Reply:
`@title={{this.closeTitle}}` ou `@title={{this.submitTitle}}`, quelle différence dans le comportement ?

Reply:
C'est une erreur, ça devrait être `@closeTitle={{this.closeTitle}}` et `@submitTitle={{this.submitTitle}}` (pas vu l'erreur étant donné que l'expect est un simple contains). Je corrige.

Reply:
Done